pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -337,11 +337,11 @@ func (m *Multicluster) checkShouldLead(client kubelib.Client, systemNamespace st
 				}
 			}
 		} else if !errors.IsNotFound(err) {
-			// TODO use a namespace informer to handle transient errors and to also allow dynamic updates
+			// A transient network error here will have a clear effect in which the remote cluster will not start
+			// working after it gets attached to a control plane. The administrator can fix the problem in this case
+			// by simply removing and then reapplying the remote secret which will presumably work the next time.
```

Comment:
Istiod may restart. That means we can have a working prod cluster, istiod scales up/restarts, and a transient error caused total production outage